travis/main.sh

```diff
@@ -40,6 +38,14 @@ check_git_status() {
 build_autotools() {
 	v --fold="bootstrap" ./bootstrap.sh
 
+	case ${TRAVIS_BUILD_TYPE} in
+	sanitizers)
+		SANITIZE_FLAGS="-fsanitize=address -fsanitize=leak -fsanitize=alignment -fsanitize=bool -fsanitize=enum -fsanitize=float-cast-overflow -fsanitize=float-divide-by-zero -fsanitize=function -fsanitize=integer-divide-by-zero -fsanitize=null -fsanitize=object-size -fsanitize=return -fsanitize=signed-integer-overflow -fsanitize=unreachable -fsanitize=unsigned-integer-overflow"
```

Comment:
:)

By maintainer mode I mean the same configure option that Werror. Does it
make more sense?
On Jun 17, 2016 10:59 AM, "Mike Frysinger" <notifications@github.com> wrote:

> In travis/main.sh
> <https://github.com/libgd/libgd/pull/235#discussion_r67458128>:
>
> > @@ -40,6 +38,14 @@ check_git_status() {
> >  build_autotools() {
> >  	v --fold="bootstrap" ./bootstrap.sh
> >
> > +	case ${TRAVIS_BUILD_TYPE} in
> > +	sanitizers)
> > +		SANITIZE_FLAGS="-fsanitize=address -fsanitize=leak -fsanitize=alignment -fsanitize=bool -fsanitize=enum -fsanitize=float-cast-overflow -fsanitize=float-divide-by-zero -fsanitize=function -fsanitize=integer-divide-by-zero -fsanitize=null -fsanitize=object-size -fsanitize=return -fsanitize=signed-integer-overflow -fsanitize=unreachable -fsanitize=unsigned-integer-overflow"
>
> hmm, making it into a general configure flag would be a good idea. that'd
> make it easier for people to run on their own system (us included). it's
> not something we'd want under maintainer mode though.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/libgd/libgd/pull/235/files/090b985e1f89a373e0a8e67f81feb79efd2a681a#r67458128>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AARPKGtBkvdg53yTdGHCatuHLLBJIsjnks5qMhuhgaJpZM4I3ZqB>
> .
>